src/taipy/core/config/_core_section.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from copy import copy
+from typing import Any, Dict, Optional
+
+from taipy.config import Config, UniqueSection
+from taipy.config._config import _Config
+from taipy.config.common._template_handler import _TemplateHandler as _tpl
+
+
+class CoreSection(UniqueSection):
+    name = "core"
+    _MODE_KEY = "mode"
+    _DEVELOPMENT_MODE = "development"
+    _EXPERIMENT_MODE = "experiment"
+    _PRODUCTION_MODE = "production"
+    _DEFAULT_MODE = _DEVELOPMENT_MODE
+    _VERSION_NUMBER_KEY = "version_number"
+    _DEFAULT_VERSION_NUMBER = ""
+    _TAIPY_FORCE_KEY = "taipy_force"
+    _DEFAULT_TAIPY_FORCE = False
+    _CLEAN_ENTITIES_KEY = "clean_entities"
+    _DEFAULT_CLEAN_ENTITIES = False
+
+    def __init__(
+        self,
+        mode: Optional[str] = None,
+        version_number: Optional[str] = None,
+        force: Optional[bool] = None,
+        clean_entities: Optional[bool] = None,
+        **properties
+    ):
+        self.mode = mode or self._DEFAULT_MODE
+        self.version_number = version_number or self._DEFAULT_VERSION_NUMBER
+        self.force = force or self._DEFAULT_TAIPY_FORCE
+        self.clean_entities = clean_entities or self._DEFAULT_CLEAN_ENTITIES
+        super().__init__(**properties)
+
+    def __copy__(self):
+        return CoreSection(self.mode, self.version_number, self.force, self.clean_entities, **copy(self._properties))
+
+    def __getattr__(self, item: str) -> Optional[Any]:
+        return _tpl._replace_templates(self._properties.get(item))
```

Comment:
`__getattr__()` method is not necessary since it is basically the same as the one implemented at `Section` level.
I will remove this.